### Задание 0 (разминка)

Даны два списка `a` и `b` (см. ниже).
    
1. Выведите на экран первый элемент списка `a`, третий элемент списка `a`, последний элемент списка `a`.
2. Добавьте в список `b` элемент 7 (просто допишите в конец).
3. Замените пятый элемент списка `a` на число 8.
4. Создайте список `merged`, который включает в себя все элементы списка `a` и списка `b`.
5. Создайте новый список `с`, который получается заменой последнего элемента списка `a` на число 100. Сам список `a` измениться не должен!

In [5]:
a = [1, 0, 9, 12, 18, 34, 89, 91, 33, 127]
b = [2, 8, 9, 11, 76, 25, 44]

In [7]:
(a[0], a[2], a[9])

(1, 9, 127)

In [8]:
b.append(7)
a[4]=8

In [9]:
merged = a + b

In [11]:
c = a
c[9] = 100


### Задание 1 (девочковое)

Есть список имен:

In [15]:
girls = ["Иветта", "Виолетта", "Кассандра", "Вирджиния", 
         "Амелия", "Розамунда", "Янина", "Беатриса"]

Используя список `girls`, выведите на экран следующие списки:

    ["Виолетта", "Кассандра", "Вирджиния", "Амелия"]
    ["Вирджиния", "Амелия", "Розамунда", "Янина", "Беатриса"]
    ["Иветта", "Виолетта", "Вирджиния", "Амелия"]
    ["Кассандра", "Амелия", "Розамунда"]

In [18]:
print(girls[1], girls[2], girls[3], girls[4])
print(girls[3], girls[4], girls[5], girls[6], girls[7])
print(girls[0], girls[1], girls[3], girls[4])
print(girls[2], girls[4], girls[5])

Виолетта Кассандра Вирджиния Амелия
Вирджиния Амелия Розамунда Янина Беатриса
Иветта Виолетта Вирджиния Амелия
Кассандра Амелия Розамунда


### Задание 2 (поэлементное) 

Дан список `L`. Выведите на экран (последовательно, с новой строчки): 

* все элементы списка `L`
* логарифмированные значения элементов списка `L`

In [19]:
L = [12, 3, 8, 125, 10, 98, 54, 199]

In [20]:
for i in L:
    print(i)
import math
for i in L:
    print(math.log(i))

12
3
8
125
10
98
54
199
2.4849066497880004
1.0986122886681098
2.0794415416798357
4.8283137373023015
2.302585092994046
4.584967478670572
3.9889840465642745
5.293304824724492


Замените пятый элемент списка `L` на 0. Проделайте те же операции, что и раньше. Объясните, почему получаются такие результаты.

In [22]:
L[4] = 0
for i in L:
    print(i)
import math
for i in L:
    print(math.log(i))

12
3
8
125
0
98
54
199
2.4849066497880004
1.0986122886681098
2.0794415416798357
4.8283137373023015


ValueError: ignored

### Задание 3 (демографическое)

В списке `age` сохранены значения возраста респондентов: 

    age = [24, 35, 42, 27, 45, 48, 33]
    
Создайте список `age2`, в котором будут храниться значения возраста, возведенные в квадрат.

**Подсказка:** используйте цикл *for*.

In [25]:
age = [24, 35, 42, 27, 45, 48, 33]
age2 = []

In [32]:
for i in age:
    age2.append(i**i)
for i in age2:
    print(i)

576
1225
1764
729
2025
2304
1089
1333735776850284124449081472843776
1102507499354148695951786433413508348166942596435546875
150130937545296572356771972164254457814047970568738777235893533016064
443426488243037769948249630619149892803
248063644451341145494649182395412689744530581492654164321720600128173828125
500702078263459319174537025249570888246709955377400223021257741084821677152403456
129110040087761027839616029934664535539337183380513
1333735776850284124449081472843776
1102507499354148695951786433413508348166942596435546875
150130937545296572356771972164254457814047970568738777235893533016064
443426488243037769948249630619149892803
248063644451341145494649182395412689744530581492654164321720600128173828125
500702078263459319174537025249570888246709955377400223021257741084821677152403456
129110040087761027839616029934664535539337183380513
1333735776850284124449081472843776
1102507499354148695951786433413508348166942596435546875
150130937545296572356771972164254457814047970568738777

### Задание 4 (игровое)

Питон просит пользователя загадать (ввести с клавиатуры) целое число $k$ от 1 до 10. Питон берет это число $k$ и выводит на экран $k$-тый элемент списка `numbers`, причем $k$-ый элемент в привычном понимании, в нумерации, которая начинается с единицы.  

Список `numbers`:

In [34]:
numbers = [1, 5, 6, 8, 10, 21, 25, 1, 0, -9, 9]

In [38]:
a = input()
a = int(a)
print(numbers[a-1])

2
5


**Пояснение:**
    
*Входные данные:*

    Введите целое число от 1 до 10: 3
    
*Выходные данные:*

    6

### Задание 5 (числовое)

У питона есть набор любимых чисел.

    favorites = [3, 7, 11, 23, 18, 48, 81]

Напишите программу, которая просит пользователя ввести целое число, и если оно нравится питону, на экран будет выводиться сообщение: "Мое любимое число!", если нет ‒ "Эх, ну почему?".

*Подсказка:* чтобы проверить, принадлежит ли некоторый объект элементам списка, можно воспользоваться оператором *in*.

In [42]:
favorites = [3, 7, 11, 23, 18, 48, 81]
a = int(input())
if a in favorites:
    print("Мое любимое число!")
else: print ("Эх, ну почему?")    

44
Эх, ну почему?


### Задание 6 (четное)

Напишите программу, которая просит пользователя ввести число и, если оно чётное, выводит на экран сообщение: "Это число чётное", если нечётное ‒ "Это число нечётное".

*Подсказка:* используйте оператор `%` для определения остатка от деления.

In [48]:
a = int(input())
if a%2 == 0:
    print("Это число четное")
else: print ("Это число нечетное")  

68
Это число четное


### Задание 7 (музыкальное)

Напишите программу, которая принимает на вход список слов такого вида:

    words = ["Speak ","to", "me ", "of", "Florence" ,"And ", "of", "the", "Renaissance"]

а возвращает список

    words_clean = ["speak", "to", "me", "of", "florence", "and", "of", "the", "renaissance"]

Другими словами, программа убирает пробелы в словах и приводит все слова к нижнему регистру. Эту задачу можно решить через циклы, но в Python есть такая полезная вещь как списковые включения. Они работают быстрее, чем циклы, и создавать новые списки на основе старых очень удобно. Прочитать про них можно [здесь](http://pythontutor.ru/lessons/lists/#section_3) и [здесь](https://nbviewer.jupyter.org/github/allatambov/Py-programming-3/blob/master/20-04/lecture-lists-contd.ipynb).

**Подсказка:** запросите `help()`  по  методам `.strip()` и `.lower()`.

In [73]:
words = ["Speak ","to", "me ", "of", "Florence" ,"And ", "of", "the", "Renaissance"]
a = [words[i].strip() for i in range(len(words))]
a = [a[i].lower() for i in range(len(a))]
a

['speak', 'to', 'me', 'of', 'florence', 'and', 'of', 'the', 'renaissance']

### Задание 8 (секретное)

Напишите программу, которая запрашивает у пользователя пароль, и далее:

- если пароль верный, выводит на экран сообщение "Login success".
- если пароль неверный, выводит на экран сообщение "Incorrect password, try again!" до тех пор, пока пользователь не введет верный пароль.

Считайте, что верный пароль программе известен, то есть уже сохранен в некоторой переменной. 

In [62]:
parol = "09876"
a = input()
if a == parol:
    print("Login success")
else: print("Incorrect password, try again!")

09876
Login success


### Задание 9 (функциональное)

Напишите функцию `nums()`, которая принимает на вход целое число, а возвращает список из двух элементов: целое число, предшествующее принятому на вход, и число, следующее за ним.

**Пример:**

*Входные данные:*

    7

*Выходные данные:*

    [6, 8]

In [68]:
def nums():
  a = int(input())
  b = a -1
  c = a + 1
  return b,c
n = nums()
n

9


(8, 10)

### Задание 10 (счетное)

Напишите функцию `my_log()`, которая принимает на вход список чисел, и возвращает список их натуральных логарифмов. Если число меньше или равно 0, на его месте в возвращаемом списке должно быть None.

**Пример:**

*Входные данные:*

    [1, 3, 2.5, -1, 9, 0, 2.71]

*Выходные данные:*

    [0.0,
     1.0986122886681098,
     0.9162907318741551,
     None,
     2.1972245773362196,
     None,
     0.9969486348916096]

In [75]:
def my_log():
  a = []
  n = 7
  for i in range(n):  
    b = float(input())
    a.append(b)
  import math
  for i in range(n):
    if a[i] > 0:
      a[i] = math.log(a[i])
    else: a[i] = "None"
  return a
b = my_log()
b


1
3
2.5
-1
9
0
2.71


[0.0,
 1.0986122886681098,
 0.9162907318741551,
 'None',
 2.1972245773362196,
 'None',
 0.9969486348916096]

### Задание 11 (зимнее)

Напишите функцию `str_lower()`, которая принимает на вход строку (набор слов через пробел), а возвращает список ее элементов в нижнем регистре. 

**Пример:**

*Входные данные:*

    "В лесу родилась ёлочка В лесу она росла"

*Выходные данные:*

    ['в', 'лесу', 'родилась', 'ёлочка', 'в', 'лесу', 'она', 'росла']
    
*Подсказка:* для разбиение строки на части, познакомьтесь с методом `.split()` – запросите по нему `help()`.

In [77]:
def str_lower():
  a = input()
  a = a.split()
  a = [a[i].lower() for i in range(len(a))]
  return a
b = str_lower()
b

В лесу родилась ёлочка В лесу она росла


['в', 'лесу', 'родилась', 'ёлочка', 'в', 'лесу', 'она', 'росла']